docs/reference/checks_config.md

```diff
@@ -0,0 +1,214 @@
+# Metadata Checks Configuration Schema
```

Comment:
*placement discussion* : would maybe place this in "Maintain and Develop" (even if it is a reference). It's not a general reference usable by many different kinds of users right? Maintainers and maybe operators only. Keeping it close to the other relevant doc makes it easier to grasp the concept in a one-stop-shop location.